x-pack/plugins/maps/public/store/util.js

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+export function copyPersistentState(input) {
+  if (typeof input !== 'object' && input !== null) {//primitive
```

Comment:
so this is wrong and should be `if (typeof input !== 'object' || input === null) `

will add unit test